src/services/webdav/dir_stream.rs

```diff
@@ -40,32 +40,41 @@ impl DirStream {
 #[async_trait]
 impl output::Page for DirStream {
     async fn next_page(&mut self) -> Result<Option<Vec<output::Entry>>> {
-        let mut oes: Vec<output::Entry> = Vec::new();
-        for _ in 0..self.size {
-            if let Some(de) = self.multistates.response.pop() {
-                let path = de.href.clone();
-                let normalized_path = &if self.root != path {
+        if self.multistates.response.is_empty() {
+            return Ok(None);
+        };
+        let oes = mem::take(&mut self.multistates.response);
+
+        let oes = oes
+            .into_iter()
+            .filter_map(|de| {
+                let path = de.href;
+                let normalized_path = if self.root != path {
                     build_rel_path(&self.root, &path)
                 } else {
                     path
                 };
 
-                if normalized_path.eq(&self.path) {
+                if normalized_path == self.path {
```

Comment:
Just a question: do we have a lint rule here? (we prefer `==` over `.eq()`)

Reply:
So far there is no such rule. But I think `a == b` is much more readable than `a.eq(&b)`.

Reply:
Got it. Thanks!

I actually used `==` first but I was worried about its real meaning, like pointer semantics in Java (comparing their memory address rather than values). So I didn't use it.